File: android/src/main/java/com/lightningkite/kwift/bluetooth/BleDevice.shared.kt
package com.lightningkite.kwift.bluetooth

import android.annotation.TargetApi
import android.bluetooth.*
import android.content.Context
import android.os.Build
import com.lightningkite.kwift.Failable
import com.lightningkite.kwift.async.DRF
import com.lightningkite.kwift.observables.EnablingEvent
import com.lightningkite.kwift.observables.Event
import com.lightningkite.kwift.observables.ObservableProperty
import com.lightningkite.kwift.observables.StandardObservableProperty
import com.lightningkite.kwift.post
import java.util.*

interface BleDevice {
    val connected: ObservableProperty<Boolean>
    val rssi: ObservableProperty<Int>
    val mtu: ObservableProperty<Int>

    fun requestMtu(mtu: Int): DRF<Unit>
    fun read(serviceId: UUID, characteristicId: UUID): DRF<ByteArray>
    fun write(serviceId: UUID, characteristicId: UUID, value: ByteArray): DRF<Unit>
    fun subscribe(
        serviceId: UUID,
        characteristicId: UUID,
        indicate: Boolean = false
    ): Event<ByteArray>
    fun close()
}

File: ios/Kwift/com/lightningkite/kwift/bluetooth/BleDevice.shared.swift
//Package: com.lightningkite.kwift.bluetooth
//Converted using Kwift2

import Foundation
import RxSwift
import RxRelay



public protocol BleDevice {
    
    var connected: ObservableProperty<Bool> { get }
    
    var rssi: ObservableProperty<Int32> { get }
    
    var mtu: ObservableProperty<Int32> { get }
    
    func requestMtu(mtu: Int32) -> Observable<Void>
    func requestMtu(_ mtu: Int32) -> Observable<Void>
    
    func read(serviceId: UUID, characteristicId: UUID) -> Observable<ByteArray>
    func read(_ serviceId: UUID, _ characteristicId: UUID) -> Observable<ByteArray>
    
    func write(serviceId: UUID, characteristicId: UUID, value: ByteArray) -> Observable<Void>
    func write(_ serviceId: UUID, _ characteristicId: UUID, _ value: ByteArray) -> Observable<Void>
    
    func subscribe(serviceId: UUID, characteristicId: UUID, indicate: Bool) -> Observable<ByteArray>
    func subscribe(_ serviceId: UUID, _ characteristicId: UUID, _ indicate: Bool) -> Observable<ByteArray>
    
    func close() -> Void
}
 
